data_reader.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+type sourceReader interface {
+	// Unmarshal unmarshals bts to the desired format
+	Unmarshal(bts []byte) (interface{}, error)
+	// IsFile checks whether str contains the file ending
+	IsFile(str string) bool
+}
+
+// readData checks if inputStrSlice provides key=value pairs and validates them using the following input techniques:
+//   * file
+//   * inline
+//   * stdin
+//
+// The "tformat" parameter defines the parser we should use to extract the data.
+func readData(inputStrSlice []string, sr sourceReader) (map[string]interface{}, error) {
+	inputCh := os.Stdin
```

Comment:
So kannst du das nur einmal benutzen. Mach ein `stdin := ioutil.ReadAll(os.Stdin)`

Reply:
I moved this part to the beginning of the function.

Reply:
After some testing, I found out that I cannot move this to the beginning of the function because `ioutil.ReadAll` blocks if os.Stdin is empty.

Reply:
I have removed the variable `inputCh` because it is not necessary to cache it again.